test/e2e/warnings_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"bytes"
+	"strconv"
+	"strings"
+	"testing"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func TestWarningsFlag(t *testing.T) {
+	minorVersion, _ := getServerMinorVersion()
+	//skip the tests if k8s version is less than 1.19 as warnings were introduced in v1.19
+	if minorVersion < 19 {
+		t.Skip("Skipping test as warnings weren't introduced before v1.19")
+	}
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, Logger{}}
+	crdName := "test-no-warnings-crd"
+	crName := "test-no-warnings-cr"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", crName})
+		kapp.Run([]string{"delete", "-a", crdName})
+	}
+
+	cleanUp()
+	defer cleanUp()
+	customWarning := "example.com/v1alpha1 CronTab is deprecated; use example.com/v1 CronTab"
+
+	crdYaml := `
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: crontabs.stable.example.com
+spec:
+  group: stable.example.com
+  versions:
+  - name: v1alpha1
+    served: true
+    storage: false
+    deprecated: true
+    deprecationWarning: "<customWarning>"
+    schema:
+      openAPIV3Schema:
+        type: object
+  - name: v1
+    served: true
+    storage: true
+    schema:
+      openAPIV3Schema:
+        type: object
+  scope: Namespaced
+  names:
+    plural: crontabs
+    singular: crontab
+    kind: CronTab
+`
+	crYaml := `
+apiVersion: "stable.example.com/v1alpha1"
+kind: CronTab
+metadata:
+  name: <cr-name>
+spec:
+  cronSpec: "* * * * */5"
+  image: my-awesome-cron-image
+`
+	testcases := []struct {
+		logMessage   string
+		yaml         string
+		options      []string
+		expected     string
+		errorMessage string
+	}{
```

Comment:
for this type of test i think its a lot less readable to do a table test than just having unrolled version.

Reply:
Makes sense. I will update it.